### PR TITLE
fix(echo): send different phase in notification when stage is skipped

### PR DESCRIPTION
**Pull request by anotherchrisberry**
Right now, if a stage is skipped, we still send a message to echo indicating it succeeded. This is really confusing if there are notifications configured.

I'm proposing we send a different, more accurate status, so we do not lose the ability to track the events, but also do not send inaccurate notifications.

### CODE REVIEW

File: orca-echo/src/main/groovy/com/netflix/spinnaker/orca/echo/spring/EchoNotifyingStageListener.groovy [hunk 1]
```diff
@@ -82,13 +82,18 @@ class EchoNotifyingStageListener implements StageListener {
     // STOPPED stages are "successful" because they allow the pipeline to
     // proceed but they are still failures in terms of the stage and should
     // send failure notifications
-    if (stage.status in [SUCCEEDED, SKIPPED]) {
+    if (stage.status == SKIPPED) {
+      log.debug("***** $stage.execution.id Echo stage $stage.name skipped v2")
+      recordEvent('stage', 'skipped', stage)
+      return
+    }
+    if (stage.status == SUCCEEDED) {
```
Comment (robfletcher) [inline orca-echo/src/main/groovy/com/netflix/spinnaker/orca/echo/spring/EchoNotifyingStageListener.groovy]: else if

File: orca-echo/src/main/groovy/com/netflix/spinnaker/orca/echo/spring/EchoNotifyingStageListener.groovy [hunk 2]
```diff
@@ -82,13 +82,18 @@ class EchoNotifyingStageListener implements StageListener {
...
       log.debug("***** $stage.execution.id Echo stage $stage.name complete v2")
       recordEvent('stage', 'complete', stage)
-    } else {
```
Comment (robfletcher) [inline orca-echo/src/main/groovy/com/netflix/spinnaker/orca/echo/spring/EchoNotifyingStageListener.groovy]: What is this, JavaScript?

File: orca-echo/src/main/groovy/com/netflix/spinnaker/orca/echo/spring/EchoNotifyingStageListener.groovy (hunk 1 above)
Comment (anotherchrisberry) [reply orca-echo/src/main/groovy/com/netflix/spinnaker/orca/echo/spring/EchoNotifyingStageListener.groovy]: fine, I'll make it look like...that

**Comment by anotherchrisberry** [comment]
@robfletcher I updated to enterprise grade `else if` statements

**Pull request merged**